/*
  MyEnglishQt Copyright Â© 2019 Juanra Goti

  This program is free software: you can redistribute it and/or modify
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation, either version 3 of the License, or
  (at your option) any later version.

  This program is distributed in the hope that it will be useful,
  but WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  GNU General Public License for more details.

  You should have received a copy of the GNU General Public License
  along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

#ifndef MAINWINDOW_H
#define MAINWINDOW_H

#include <QMainWindow>
#include <QSqlDatabase>

namespace Ui {
class MainWindow;
}

class MainWindow : public QMainWindow
{
    Q_OBJECT

public:
    explicit MainWindow(QWidget *parent = nullptr);
    ~MainWindow();

private slots:
    void on_pushButtonAbout_clicked();
    void on_pushButtonExit_clicked();
    void on_pushButtonAlfabeto_clicked();
    void on_pushButtonExanGrammar_clicked();
    void on_pushButtonEditGrammar_clicked();
    void on_pushButtonHelp_clicked();

private:
    QSqlDatabase db;

    void funConnectBD(QString fileDb);
    void funCreateDb(QString fileDb);

    Ui::MainWindow *ui;
};

#endif // MAINWINDOW_H
